pkg/imgpkg/registry/keychain_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package registry_test
+
+import (
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/k14s/imgpkg/pkg/imgpkg/registry"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAnonAuthWhenNoEnvVarsProvided(t *testing.T) {
```

Comment:
Please refactor these tests to use the `t.Run` format that we have been using in `imgpkg`

Reply:
Good idea. Also added a parent test context to test the rules around ordering around using multiple keychains